pkg/util/azureclient/mgmt/redhatopenshift/2023-04-01/redhatopenshift/openshiftclusters_addons.go

```diff
@@ -0,0 +1,87 @@
+package redhatopenshift
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	mgmtredhatopenshift20230401 "github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/2023-04-01/redhatopenshift"
+)
+
+// OpenShiftClustersClientAddons contains addons for OpenShiftClustersClient
+type OpenShiftClustersClientAddons interface {
+	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230401.OpenShiftCluster) error
+	DeleteAndWait(ctx context.Context, resourceGroupName string, resourceName string) error
+	List(ctx context.Context) (clusters []mgmtredhatopenshift20230401.OpenShiftCluster, err error)
+	ListByResourceGroup(ctx context.Context, resourceGroupName string) (clusters []mgmtredhatopenshift20230401.OpenShiftCluster, err error)
+	UpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230401.OpenShiftClusterUpdate) (result mgmtredhatopenshift20230401.OpenShiftCluster, err error)
+}
+
+func (c *openShiftClustersClient) CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230401.OpenShiftCluster) error {
+	future, err := c.CreateOrUpdate(ctx, resourceGroupName, resourceName, parameters)
+	if err != nil {
+		return err
+	}
+
+	return future.WaitForCompletionRef(ctx, c.Client)
+}
```

Comment:
is this used anywhere? I have not found the references to that function apart from its definition and the interface definition. (although I have only done a search on the pr so I could have missed it)

The others seem to be also not used in files in this pr?

Reply:
This client is used in E2E tests, specifically in `test/e2e/tags.go`. I copied most of the code for this package over from the corresponding one for the `2022-09-04` API version, though I added the `UpdateAndWait()` function.